src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/PrisonApiClient.kt

```diff
@@ -164,13 +170,15 @@ class PrisonApiClient(@Qualifier("prisonApiWebClient") private val webClient: We
       .uri("/api/offenders/$offenderNo/booking")
       .bodyValue(detail)
       .retrieve()
-      .bodyToMono(InmateDetail::class.java)
-      .onErrorResume(WebClientResponseException::class.java) {
-        propagateClientError(
-          it,
-          "Client error when posting to /api/offenders/$offenderNo/booking"
-        )
+      .onStatus(HttpStatus::is4xxClientError) { response ->
+        response.bodyToMono(ErrorResponse::class.java).map {
```

Comment:
could extract 
```
{ response ->
        response.bodyToMono(ErrorResponse::class.java).map {
          ClientException(
            it,
            "Client error when posting to /api/offenders/$offenderNo/recall"
          )
        }
      }
```

into a helper method, could event use the same name? `propagateClientError`